cmd/snap-confine/mount-support.c

```diff
@@ -581,7 +581,21 @@ void sc_populate_mount_ns(struct sc_apparmor *apparmor, int snap_update_ns_fd,
 					die("cannot locate the core or legacy core snap (current symlink missing?)");
 				}
 			}
-			die("cannot locate the base snap: %s", base_snap_name);
+			if (sc_streq(base_snap_name, "core16")) {
```

Comment:
I would rather do this in snapd, not in snap-confine.

Reply:
Yeah, I was thinking about this too, I mostly did it this way because we already have compat code for ubuntu-core->core in snap-confine and also because it feels more isolated. But I don't have super strong opinions.

Reply:
notice if I undertand correctly that done this way it means we pivot but to core (not use the legacy mode), this feels correct, it also means any bugs about interfaces and mounting for core18 will also show up for snaps moving to base: core16.

If we do something in cmd_run.go it should also result in a pivot!

Reply:
Yeah, I would rather keep core16 and core as similar as possible. It's much better to transition to something safe in one step and then venture into the unknown rather than take both jumps at once.

Reply:
That's not possible because we cannot make that true on a core 18 device, I fear things are already confused there. Do we pivot no base snap on a core 18 device, or it sees the root filesystem (!= core/core16) ?

Reply:
we would pivot if the base is core16 (before being translated), the check is like this:

we do pivot if

bool sc_should_use_normal_mode(sc_distro distro, const char *base_snap_name)
{
	return distro != SC_DISTRO_CORE16 || !sc_streq(base_snap_name, "core");
}